pkg/controllers/clusterresourceplacement/resource_selector.go

```diff
@@ -131,38 +169,56 @@ func (r *Reconciler) gatherSelectedResource(placement string, selectors []fleetv
 
 func sortResources(resources []*unstructured.Unstructured) {
 	sort.Slice(resources, func(i, j int) bool {
+		// build the ordering map.
+		ordering := make(map[string]int, len(ApplyOrder))
```

Comment:
Hi Arvind! This is re-generating the same static content upon every call if I understand it correctly?

Reply:
This might not be necessary and can be added as a top-level variable upon package initialization.